subsys/bluetooth/host/gatt.c

```diff
@@ -118,10 +126,13 @@ static ssize_t read_ppcp(struct bt_conn *conn, const struct bt_gatt_attr *attr,
 		uint16_t timeout;
 	} ppcp;
 
-	ppcp.min_int = sys_cpu_to_le16(CONFIG_BT_PERIPHERAL_PREF_MIN_INT);
-	ppcp.max_int = sys_cpu_to_le16(CONFIG_BT_PERIPHERAL_PREF_MAX_INT);
+	ppcp.min_int = sys_cpu_to_le16(CONFIG_BT_PERIPHERAL_PREF_MIN_INT ? :
+				       0xffff);
```

Comment:
Here's the code-size & runtime overhead I mentioned. If you just allow 0xffff directly in Kconfig you will only insure an overhead for build time but not runtime, and no code size overhead either. You might want to consider changing the Kconfig types from `int` to `hex` if you think it's unfriendly to require the user to specify the decimal value instead of 0xffff.

Reply:
Follow-up: most likely the compiler will optimize away the never-taken branch, however I still think it's cleaner if right value comes straight from Kconfig